sequence_processing_pipeline/Job.py

```diff
@@ -191,20 +212,81 @@ def _system_call(self, cmd, allow_return_codes=[], callback=None):
 
         return {'stdout': stdout, 'stderr': stderr, 'return_code': return_code}
 
+    def wait_on_job_ids(self, job_ids, callback=None):
+        '''
+        Wait for the given job-ids to finish running before returning.
+        :param job_ids: A list of Slurm job-ids
+        :param callback: Set callback function that receives status updates.
+        :return: A dictionary of job-ids and their current statuses.
+        '''
+
+        # wait_on_job_ids was broken out of submit_job() and updated to monitor
+        # multiple job ids. This will allow multiple jobs to be submitted to
+        # Slurm in parallel and a single wait_on_job_ids() can wait on all of
+        # them before returning, optionally submitting callbacks for each
+        # job-id.
+
+        def query_slurm(job_ids):
+            # internal function query_slurm encapsulates the handling of
+            # squeue.
+            result = self._system_call(f"squeue -t all -j {','.join(job_ids)}"
+                                       " -o '%F,%A,%T'")
+
+            if result['return_code'] != 0:
+                raise ExecFailedError(result['stderr'])
```

Comment:
What about having a wait/try for 3 times before raising an error; things can fail for many reasons.

Reply:
okay. I'll try adding a sleep and repeat to code.  testing now on qiita-rc.